source/fundamentals/connection/socks.txt

```diff
@@ -0,0 +1,139 @@
+.. _node-connect-socks:
+
+===========================
+Enable SOCKS5 Proxy Support
+===========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to connect to MongoDB instances with
+SOCKS5 proxy support. SOCKS5 is a standardized protocol for connecting
+to network services through a separate proxy server.
+
+.. tip::
+   
+   To learn more about SOCKS, see the Wikipedia entry on
+   :wikipedia:`SOCKS <SOCKS>`.
+
+Load the socks Module
+---------------------
+
+Starting in version 6.0 of the {+driver-short+}, you must *manually* load
+the ``socks`` dependency to use SOCKS5 proxy support in your
+application. You can install ``socks`` by running the following command
+in your terminal:
+
+.. code-block:: bash
+   
+   npm i socks
+
+SOCKS5 Client Options
+---------------------
+
+You can set options in your ``MongoClientOptions`` instance or
+parameters in your connection URI to configure SOCKS5 proxy support for 
+your connection. The following table describes the client options
+related to SOCKS5.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1 1
+
+   * - Name
+     - Accepted Values
+     - Default Value
+     - Description
+
+   * - **proxyHost**
+     - string
+     - ``null``
+     - Specifies the IPv4/IPv6 address or domain name on which a SOCKS5
+       proxy is listening.
+
+   * - **proxyPort**
+     - non-negative integer
+     - ``null``
+     - Specifies the TCP port number of the SOCKS5 proxy server specified in the
+       ``proxyHost`` option. If you have set the ``proxyHost`` option,
+       this value of this option defaults to ``1080``.
+
+   * - **proxyUsername**
+     - string
+     - ``null``
+     - Specifies the username for username and password authentication to the SOCKS5
+       proxy server specified in the ``proxyHost`` option. If you set
```

Comment:
Suggestion:

I think it's important to identify it's for the proxy server, but think it's reasonable to expect the user to understand that the proxy is the one specified in the proxyHost option.

```suggestion
       proxy server. If you set
```